File: Algorithm/392. Is Subsequence.py
# -*- coding: utf-8 -*-
# @Author: WuLC
# @Date:   2016-09-09 21:06:33
# @Last modified by:   WuLC
# @Last Modified time: 2016-09-09 21:06:40
# @Email: liangchaowu5@gmail.com

class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        if len(t) == 0:
            return False
        index = 0
        for char in t:
            if s[index] == char:
                index += 1
            if index == len(s):
                return True
        return False
        

File: Algorithm/Java/392. Is Subsequence.java
/*
 * Created on Sat Apr 07 2018 17:30:52
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

// simple solution, O(n) time, n is the length of t
class Solution 
{
    public boolean isSubsequence(String s, String t) 
    {
        if (s.length() == 0) return true;
        int idx = 0;
        for(int i=0; i<t.length(); i++)
        {
            if(t.charAt(i) == s.charAt(idx)) idx++;
            if(idx == s.length()) return true;
        }
        return false;
    }
}